api/jobset/v1alpha2/jobset_types.go

```diff
@@ -218,10 +222,52 @@ const (
 	OperatorAny Operator = "Any"
 )
 
+// FailurePolicyAction defines the action the JobSet controller will take for
+// a given FailurePolicyRule.
+type FailurePolicyAction string
+
+const (
+	// Fail the JobSet immediately, regardless of maxRestarts.
+	FailJobSet FailurePolicyAction = "FailJobSet"
+
+	// Restart the JobSet if the number of restart attempts is less than MaxRestarts.
+	// Otherwise, fail the JobSet.
+	RestartJobSet FailurePolicyAction = "RestartJobSet"
+
+	// Do not count the failure against maxRestarts.
+	RestartJobSetAndIgnoreMaxRestarts FailurePolicyAction = "RestartJobSetAndIgnoreMaxRestarts"
```

Comment:
It would be ideal to have a similar length. Looking at PodFailurePolicy, we have FailJob, CountJob or Ignore.

Maybe we could try  IgnoreMaxRestarts?

Reply:
I am not particularly tied with the name. However, do we know if there was a specific reason the [KEP](https://github.com/kubernetes-sigs/jobset/tree/6a74ec27dcff34d6862d24fc7936fc7ef0c89be2/keps/262-ConfigurableFailurePolicy) went with `RestartJobSetAndIgnoreMaxRestarts`?

I do agree it would be ideal if the names have similar lengths. However, a nice quality of the current name `RestartJobSetAndIgnoreMaxRestarts` is that it makes it clear how this action is related to the `RestartJobSet` action. It would be great if whatever we changed the name to maintained this property.

Reply:
I forgot that this was the KEP.

Reply:
Do we want to continue discussing using a different name or can this comment chain be resolved? @kannon92 and @danielvegamyhre

Reply:
I like the more explicit name. It would be prettier if they were all similar lengths but that is a secondary concern imo.